src/named-groups/clients/NamedGroupsClient.ts

```diff
@@ -0,0 +1,169 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { NAMED_GROUPS_BASE_PATH, OperationsBase } from "../../common/OperationsBase";
+import { EntityListIterator } from "../../common/iterators/EntityListIterator";
+import { EntityListIteratorImpl } from "../../common/iterators/EntityListIteratorImpl";
+import { EntityCollectionPage, getEntityCollectionPage } from "../../common/iterators/IteratorUtil";
+import { RequiredError } from "../../common/Errors";
+import { AccessToken } from "@itwin/core-bentley";
+import { PreferReturn } from "../../common/Common";
+import { NamedGroup, NamedGroupContainer, NamedGroupCreate, NamedGroupList, NamedGroupMetadata, NamedGroupMinimal, NamedGroupMinimalList, NamedGroupUpdate } from "../interfaces/NamedGroups";
+import { INamedGroupsClient } from "../interfaces/INamedGroupsClient";
+
+export class NamedGroupsClient extends OperationsBase implements INamedGroupsClient {
+  private _baseUrl = `${this.basePath}`;
+  private static readonly MAX_DISPLAY_NAME_LENGTH = 512;
+
+  constructor(basePath?: string) {
+    super(basePath ?? NAMED_GROUPS_BASE_PATH);
+  }
+
+  public async createNamedGroup(accessToken: AccessToken, group: NamedGroupCreate): Promise<NamedGroup> {
+    if (this.isNullOrWhitespace(group.query)) {
+      throw new RequiredError(
+        "query",
+        "Required field query was null or undefined.",
+      );
+    }
+
+    if (!this.isWithinMaxAllowedCharacters(group.displayName, NamedGroupsClient.MAX_DISPLAY_NAME_LENGTH)) {
+      throw new RequiredError(
+        "displayName",
+        "Field displayName was invalid. It must be a string with a maximum length of 512 characters.",
+      );
+    }
+
+    group.metadata && this.validateMetadata(group.metadata);
+
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(group));
+    return (await this.fetchJSON<NamedGroupContainer>(this._baseUrl, requestOptions)).group;
+  }
+
+  public async deleteNamedGroup(accessToken: AccessToken, groupId: string): Promise<Response> {
+    const url = this.constructUrl(groupId);
+    const requestOptions: RequestInit = this.createRequest("DELETE", accessToken);
+    return this.fetchJSON<Response>(url, requestOptions);
+  }
+
+  public async updateNamedGroup(accessToken: AccessToken, groupId: string, group: NamedGroupUpdate): Promise<NamedGroup> {
+    if (null == group.displayName && null == group.description && null == group.query) {
```

Comment:
1. metadata check is missing
2. Wouldn't `{}` result in all of these properties being `undefined` instead of `null`? I think you should be checking for both.

Reply:
Checking for both now.

Reply:
The changed implementation does not work well with empty strings.

Reply:
Changed it so an empty `description` is considered valid now.